src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AspNetCore/GenericHostWebHostBuilderExtensionsWrapper.cs

```diff
@@ -12,6 +12,7 @@ namespace NewRelic.Providers.Wrapper.AspNetCore
     public class GenericHostWebHostBuilderExtensionsWrapper : IWrapper
     {
         public bool IsTransactionRequired => false;
+        private static bool _isNewRelicStartupFilterAdded;
```

Comment:
Does this need to be `volatile` or do we need to use a `Interlocked.CompareExchange`? To answer this question we need to know if this wrapper is ever executed concurrently.

Reply:
I don't know for sure if this wrapper will be ever executed concurrently as both of the `ConfigureWebHost` method overloads are public. At the current state of the aspnetcore repo, the 2 overloads are executed one after another as the result of one calling another.

https://github.com/Tratcher/aspnetcore/blob/master/src/Hosting/Hosting/src/GenericHostWebHostBuilderExtensions.cs

Reply:
I'm fine with keeping the solution simpler for now.